feat: add notification nudge component in video upload + feat: new fcm notification flow

joel-medicala-yral opened the pull request:
closes https://github.com/dolr-ai/product-roadmap/issues/329
closes https://github.com/dolr-ai/product-roadmap/issues/330

uwuxyz committed:
feat: add notification nudge component and integrate with video upload flow

- Introduced a new `NotificationNudge` component to prompt users for enabling notifications.
- Updated the `VideoUploader` component to display the notification nudge when a video is being processed.
- Refactored notification handling logic to utilize the new `get_device_registeration_token` function for device registration.
- Updated dependencies for `yral-metadata-client` and `yral-metadata-types` in `Cargo.toml` and `Cargo.lock`.

uwuxyz committed:
refactor: update Firebase messaging integration and clean up notification handling

- Migrated to Firebase SDK version 10.14.1, replacing compat imports with modular imports.
- Implemented foreground message handling using a custom event for Leptos.
- Removed unnecessary `send_wrap` from `get_device_registeration_token` calls.
- Cleaned up notification handling logic in the service worker and client-side code.
- Updated `Cargo.toml` to reflect changes in `yral-metadata-client` and `yral-metadata-types` dependencies.

uwuxyz committed:
refactor: enhance Firebase messaging integration and streamline notification handling

- Updated service worker to use Firebase compat SDK for messaging.
- Improved initialization logic for Firebase to ensure proper app instance usage.
- Refactored notification handling in the service worker for better clarity and error logging.
- Simplified action handlers in notification components by removing unnecessary error handling.
- Updated client-side token retrieval to improve error handling and logging.

uwuxyz committed:
refactor: streamline notification action handlers for improved clarity

- Removed unnecessary `send_wrap` import from notification components.
- Simplified action handlers in `NotifInnerComponent` and `EnableNotifications` functions for better readability.
- Ensured consistent handling of device registration and unregistration logic.

uwuxyz committed:
fix: fixed sw(s) + feat: added sample notification components with integration for future use

ravi-sawlani-yral commented on Cargo.toml:
```diff
@@ -142,8 +142,8 @@ yral-canisters-common = { git = "https://github.com/yral-dapp/yral-common.git",
 ciborium = "0.2.2"
-yral-metadata-client = { git = "https://github.com/yral-dapp/yral-metadata", rev = "56e3f1f1f5f452673bee17739520c800c1264295" }
-yral-metadata-types = { git = "https://github.com/yral-dapp/yral-metadata", rev = "56e3f1f1f5f452673bee17739520c800c1264295" }
+yral-metadata-client = { git = "https://github.com/yral-dapp/yral-metadata", rev = "a641efec208088d940df2eafec11a23595217694" }
+yral-metadata-types = { git = "https://github.com/yral-dapp/yral-metadata", rev = "a641efec208088d940df2eafec11a23595217694" }
```
use main branch. We are not using pinned commits anywhere.

ravi-sawlani-yral commented on ssr/public/firebase-messaging-sw.js:
```diff
@@ -1,18 +1,94 @@
+
+const firebaseConfig = {
+
+  apiKey: "AIzaSyCc_3-30sOgNhpPprV-YDMSTebf4EAPNIo",
```
is this not private?

uwuxyz committed:
refactor: update yral-metadata dependencies and improve notification handling

- Updated `yral-metadata-client` and `yral-metadata-types` dependencies to a new commit for enhanced functionality.
- Refactored device registration logic in notification components to use `delegate_identity` for improved identity handling.
- Removed the unused `getDeviceFingerprint` function from the notification utility, streamlining the codebase.

joel-medicala-yral committed:
Merge branch 'main' into joel/notif-front

uwuxyz committed:
Update dependencies in Cargo.lock and Cargo.toml

- Bump versions for several packages including `bitflags`, `candid`, `cc`, `errno`, `windows-core`, `windows-result`, and `windows-strings` to their latest releases.
- Remove unused `yral-pump-n-dump-common` dependency from Cargo.toml to streamline the project.

uwuxyz committed:
Update notification body handling in firebase-messaging-sw.js

- Enhanced the notification body to prioritize the message field if available, falling back to the default message if not. This improves the clarity of notifications sent to users.

uwuxyz committed:
lint

uwuxyz committed:
lint

komal-sai-yral commented on ssr/src/page/src/settings/mod.rs:
```diff
@@ -117,20 +121,41 @@ fn ProfileInfo() -> impl IntoView {
     let on_token_click: Action<(), (), LocalStorage> = Action::new_unsync(move |()| async move {
-        get_token_for_principal(user_details.principal.to_string()).await;
+        let metaclient: MetadataClient<false> = MetadataClient::with_base_url(
+            reqwest::Url::parse("https://pr-19-dolr-ai-yral-metadata.fly.dev/").unwrap(),
```
add this to const

komal-sai-yral commented on ssr/src/page/src/upload/video_upload.rs:
```diff
@@ -344,6 +345,7 @@ pub fn VideoUploader(
 
     view! {
         <div class="flex flex-col-reverse lg:flex-row w-full gap-4 lg:gap-20 mx-auto justify-center items-center min-h-screen bg-transparent p-0">
+            <NotificationNudge pop_up=notification_nudge />
```
this means notification will only be shown if we are on the same page right ? is this expected ?

komal-sai-yral reviewed (commented):
minor comments. 

one question - how are we handling token expiry ?

uwuxyz committed:
Update yral-metadata dependencies and refactor notification registration logic

- Updated `yral-metadata-client` and `yral-metadata-types` to a new commit for improved functionality.
- Refactored device registration logic in notification components to directly use `cans.identity()` instead of `delegate_identity`, simplifying the code.

joel-medicala-yral committed:
Merge branch 'main' into joel/notif-front

uwuxyz committed:
Refactor notification handling and simplify MetadataClient initialization

- Updated notification handling in `firebase-messaging-sw.js` to prioritize data fields for title and body, enhancing user clarity.
- Simplified `MetadataClient` initialization across multiple components by using the default constructor instead of specifying a base URL, improving maintainability.

uwuxyz committed:
Update yral-metadata dependencies to use the master branch

- Changed `yral-metadata-client` and `yral-metadata-types` dependencies in both `Cargo.lock` and `Cargo.toml` to point to the master branch instead of specific commits, ensuring the latest updates are included.

uwuxyz committed:
fixed: notif nudge

uwuxyz committed:
fix: fixed double notification issue

uwuxyz committed:
feat: enhance notification permission handling and registration logic

- Added `get_notification_permission` utility to check and handle notification permissions before device registration.
- Updated notification handling in `NotificationNudge`, `EnableNotifications`, and `NotificationWalletImpl` to ensure proper permission checks.
- Refactored notification state management to improve clarity and maintainability across components.

uwuxyz committed:
refactor: improve notification permission checks and code clarity

- Consolidated notification permission checks across `NotificationNudge`, `EnableNotifications`, and `NotificationWalletImpl` for consistency.
- Enhanced readability by restructuring conditional logic in notification handling.
- Removed redundant imports to streamline the codebase.

uwuxyz committed:
refactor: remove unused notification permission import

- Eliminated the unused `get_notification_permission` import from the notification module to streamline the code and improve clarity.

uwuxyz committed:
fixed edge cases

uwuxyz committed:
chore: updated deps to main

uwuxyz committed:
fix: race condition & notification dedupe

uwuxyz committed:
fix

uwuxyz committed:
fix: race condition

uwuxyz committed:
fix: race condition

uwuxyz committed:
fix: race condition

uwuxyz committed:
feat: integrate local storage for notification settings

- Added local storage management for notification settings in GoogleAuthProvider, LocalStorageProvider, and Logout components.
- Ensured notifications are disabled upon login and logout actions to streamline user experience.

uwuxyz committed:
pr url

uwuxyz committed:
lint

uwuxyz committed:
feat: enhance notification nudge with session management

- Introduced local storage for tracking the last interaction time.
- Added logic to determine if the user is in a new session based on the last interaction timestamp.
- Updated notification nudge visibility conditions to include session state.

saikatdas0790 committed:
Merge branch 'main' into joel/notif-front

uwuxyz committed:
pointitng to prod

uwuxyz committed:
lint

joel-medicala-yral committed:
Merge branch 'main' into joel/notif-front

uwuxyz committed:
fix

uwuxyz committed:
feat: register service worker for Firebase messaging

- Added logic to check and register a service worker for Firebase messaging.
- Included console logs for successful registration and error handling.

uwuxyz committed:
refactor: simplify notification nudge logic

- Removed session management related to last interaction time and session timeout.
- Updated notification visibility conditions to streamline functionality.

uwuxyz committed:
refactor: remove service worker registration from Firebase messaging setup

- Eliminated the service worker registration logic from the Firebase messaging initialization function.
- Updated console logs to reflect the removal of service worker checks and registration.

uwuxyz committed:
refactor: clean up Firebase messaging service worker and initialization logic

- Removed commented-out background message handling code from the service worker.
- Simplified console log messages for Firebase initialization.
- Ensured Firebase is initialized before requesting FCM tokens in the notification setup.
- Updated token retrieval logs for clarity.

uwuxyz committed:
refactor: clean up notification display logic

- Removed commented-out code related to the Notification struct path.
- Simplified the reset of the slide-out state in the notification display function.

uwuxyz committed:
refactor: clean up Firebase messaging and notification utility code

- Removed commented-out code and unnecessary comments in the Firebase messaging service worker and notification utility functions.
- Streamlined the notification permission checks and token retrieval logic for clarity and maintainability.
